yake/src/main.rs

```diff
@@ -0,0 +1,107 @@
+use std::path::PathBuf;
+
+use clap::{Parser, ValueEnum};
+use yake_rust::{Config, StopWords, Yake};
+
+#[derive(Clone, Copy, ValueEnum)]
+enum DedupFunction {
+    Level,
+    Jaro,
+    SeqM,
+}
+
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    // -ti, --text_input TEXT          Input text, SURROUNDED by single quotes(')
+    #[arg(conflicts_with = "input_file", long)]
+    text_input: String,
+
+    // -i, --input_file TEXT           Input file
+    #[arg(conflicts_with = "text_input", short, long)]
+    input_file: PathBuf,
+
+    // -n, --ngram-size INTEGER        Max size of the ngram.
+    #[arg(short, long)]
+    ngram_size: usize,
+
+    // -df, --dedup-func [leve|jaro|seqm]
+    // 								Deduplication function.
+    #[arg(value_enum, long)]
+    dedup_func: DedupFunction,
+
+    // -dl, --dedup-lim FLOAT          Deduplication limiar.
+    #[arg(long, value_parser = dedup_parser)]
+    dedup_lim: f64,
+
+    // -ws, --window-size INTEGER      Window size.
+    #[arg(long)]
+    window_size: usize,
+
+    // -t, --top INTEGER               Number of keyphrases to extract
+    #[arg(short, long, default_value = "10")]
+    top: u32,
```

Comment:
This argument is really optional as the person might request the full list

Reply:
Optional.